vmmlib/vector.hpp

```diff
@@ -68,27 +68,29 @@ template< size_t M, typename T > class vector
 
 #ifndef SWIG
     // initializes the first M-1 values from vector_, the last from last_
-    vector( const vector< M-1, T >& vector_, T last_ );
+    template< typename TT >
+    vector( const vector< M-1, TT >& vector_, T last_,
+            typename enable_if< M == 4, TT >::type* = nullptr );
 #endif
 
     explicit vector( const T* values );
 
 #ifdef __OSG_MATH
     template< typename OSGVEC3 >
     explicit vector( const OSGVEC3& from,
-                     typename enable_if< M == 3, OSGVEC3 >::type* = 0 );
+                     typename enable_if< M == 3, OSGVEC3 >::type* = nullptr );
 #endif
 
     // vec< M > with homogeneous coordinates <-> vec< M-1 > conversion ctor
     // to-homogenous-coordinates ctor
-    template< size_t N >
-    vector( const vector< N, T >& source_,
-            typename enable_if< N == M - 1 >::type* = 0 );
+    template< typename TT >
+    vector( const vector< 3, TT >& source_,
+            typename enable_if< M == 4, TT >::type* = nullptr );
```

Comment:
I am not sure if the enable_if is needed, homogeneous coordinates are not only defined for vectors 3/4.